Encryption.md

```diff
@@ -19,216 +19,544 @@
-required to protect the column-specific information in the file footer. In these cases, 
-the column-specific information (kept in `ColumnMetaData` structures) is split from the 
-footer, by utilizing the `required i64 file_offset` parameter in the `ColumnChunk` 
-structure. This allows to serialize each `ColumnMetaData` structure separately, and encrypt 
-it with a column-specific key, thus protecting the column stats and other metadata. 
-
-### Encrypted footer mode
-
+attacker (e.g. schema, num_values, key_value_metadata, encoding 
+and crypto_meta_data). This information is automatically protected when the footer and 
+secret columns are encrypted with the same key. In other cases - when column(s) and the 
+footer are encrypted with different keys; or column(s) are encrypted and the footer is not, 
+an extra measure is required to protect the column-specific information in the file footer. 
+In these cases, the `ColumnMetaData` structures are Thrift-serialized separately and encrypted 
+ with the AES GCM cipher and a column-specific key, thus protecting the column stats and 
+ other metadata. The encryption buffer is kept in an `optional binary encrypted_column_metadata` 
+ field in the `ColumnChunk`.
+
+```c
+struct ColumnChunk {
+...
+  
+  /** Column metadata for this chunk.. **/
+  3: optional ColumnMetaData meta_data
+..
+
+  /** Encrypted column metadata for this chunk **/
+  9: optional binary encrypted_column_metadata
+}
+```
+
+
+### 5.2 Encrypted footer mode
 In files with sensitive column data, a good security practice is to encrypt not only the 
-secret columns, but also the file footer metadata, with a separate footer key. This hides
-the file schema / column names, number of rows, key-value properties, column sort order, 
-column data offset and size, list of encrypted columns and metadata of the column encryption keys. 
-It also makes the footer tamper-proof.
-
-The columns encrypted with the same key as the footer, don't split the ColumnMetaData from the 
-ColumnChunks, leaving it at the original location, `optional ColumnMetaData meta_data`. This field
-is not set for columns enrypted with a column-specific key.
-
-A Thrift-serialized `FileCryptoMetaData` structure is written after the footer. It contains 
-information on the file encryption algorithm and on the footer (offset in 
-the file, and an optional key metadata, with a maximal length of 512). Then 
-the length of this structure is written, as a 4-byte little endian integer. Then a final 
-magic string, "PARE".
-
-Only the `FileCryptoMetaData` is written as a plaintext, all other file parts are protected
-(as needed) with appropriate keys.
-
+secret columns, but also the file footer metadata. This hides the file schema, 
+number of rows, key-value properties, column sort order, names of the encrypted columns  
+and metadata of the column encryption keys. 
+
+The columns encrypted with the same key as the footer leave the column metadata at the original 
```

Comment:
To be clear, can you update this to "must leave"? I don't want this interpreted as optional when the key is identical.